packages/caravan-fees/src/rbf.ts

```diff
@@ -0,0 +1,303 @@
+import { PsbtV2 } from "@caravan/psbt";
+import BigNumber from "bignumber.js";
+import { DEFAULT_DUST_THRESHOLD, RBF_SEQUENCE } from "./constants";
+import { RbfOptions, FeeRate } from "./types";
+
+/**
+ * RbfTransaction class for handling Replace-By-Fee transactions
+ */
+class RbfTransaction {
+  private psbt: PsbtV2;
+  private newFeeRate: FeeRate;
+  private dustThreshold: string;
+
+  /**
+   * Create a new RbfTransaction
+   * @param psbt - The original transaction as a PsbtV2
+   * @param newFeeRate - The new fee rate for the replacement transaction
+   * @param options - Additional options for RBF
+   *   - dustThreshold: The minimum amount (in satoshis) for an output to not be considered dust.
+   *If not provided, a default value is used.
+   */
+  constructor(
+    psbt: PsbtV2,
+    newFeeRate: FeeRate,
+    options?: { dustThreshold?: string }
```

Comment:
Why not just have one options object that has dust, fee rate and psbt?

Reply:
well the idea was to include more option fields in future , maybe related to fee-bumping priority or something ... so that's why I kept it like that

Reply:
Done [here](https://github.com/caravan-bitcoin/caravan/commit/77f0cbeb67a274b0aea5a69e87168b9ac1e2304a#diff-47122bb461c82a41c84d8866adf0cb005bf7f573a2afb8558f670a419bed146aR50)

[RBFOPTIONS](https://github.com/caravan-bitcoin/caravan/commit/77f0cbeb67a274b0aea5a69e87168b9ac1e2304a#diff-593f8d602f801c56261ed7a8dc941947102f6a7b26c483c5ee8cf9c293fffb33R41)